OpenKeychain/src/main/java/org/sufficientlysecure/keychain/remote/ui/RemoteSelectPubKeyActivity.java

```diff
@@ -103,7 +103,7 @@ protected void onCreate(Bundle savedInstanceState) {
                     public void onClick(View v) {
                         // add key ids to params Bundle for new request
                         Intent resultData = extras.getParcelable(EXTRA_DATA);
-                        resultData.putExtra(OpenPgpApi.EXTRA_KEY_IDS,
+                        resultData.putExtra(OpenPgpApi.EXTRA_KEY_IDS_SELECTED,
```

Comment:
This is the principal change here: By using a different extra, we know that the disambiguation happened (and what its result was), and can use its result instead of calling `getKeyIdsFromEmails`. An alternative would be to just return a boolean and adding the keys to `EXTRA_KEY_IDS`, but that just muddles input we received from the client app and our own state together.